### PR TITLE
fix expression enclosed in parentheses

### LINKED ISSUE #363: expression enclosed in parentheses has no effect in return block
gopher-lua v0.0.0-20210529063254-f4c35e4016d9

From lua [doc](https://www.lua.org/manual/5.1/manual.html#2.5):
`Any expression enclosed in parentheses always results in only one value. Thus, (f(x,y,z)) is always a single value, even if f returns several values. (The value of (f(x,y,z)) is the first value returned by f or nil if f does not return any values.)
`

This code will fail with gopher-lua:
```lua
local function mult_ret()
   return 1, 2
end

local function enclosed_ret()
   return (mult_ret())
end

a,b = enclosed_ret()
assert(a == 1)
assert(b == nil) -- fails here, b = 2
```
In default lua-5.1 all is OK:
```
Lua 5.1.5  Copyright (C) 1994-2012 Lua.org, PUC-Rio
> function mult_ret()
   return 1, 2
end
> function enclosed_ret()
   return (mult_ret())
end
> a,b = enclosed_ret()
> assert(a == 1)
> assert(b == nil)
> 
```

### PR DESCRIPTION
**Pull request by 0x501D**
Expression enclosed in parentheses has no effect in return block.
From lua doc:
Any expression enclosed in parentheses always results in only one value.
Thus, (f(x,y,z)) is always a single value, even if f returns several values.
(The value of (f(x,y,z)) is the first value returned by f or nil if f does
not return any values.)

https://www.lua.org/manual/5.1/manual.html#2.5

This code will fail with gopher-lua:

```lua
-- example 1:
local function mult_ret()
   return 1, 2
end

local function enclosed_ret()
   return (mult_ret())
end

a,b = enclosed_ret()
assert(a == 1)
assert(b == nil) -- fails: b = 2

-- example 2:
local function enclosed_vararg(...) return (...) end
a,b,c=enclosed_vararg(1, 2, 3)
assert(a == 1 and b == nil and c == nil) -- fail: b = 2, c = 3

-- example 3:
local function enclosed_assign(...)
    local a,b = (...)
    return a,b
end
a,b,c=enclosed_assign(1, 2, 3)
assert(a == 1 and b == nil and c == nil) -- fail: b = 2, c = 3
```
Closes #363

### CODE REVIEW

File: _lua5.1-tests/calls.lua [hunk 1]
```diff
@@ -215,6 +215,14 @@ assert(a==1 and b==1 and c==nil and d==nil)
...
 assert(a[1] == 1 and a[2] == 3 and a[3] == "a" and a[4] == "b")
 
+-- Any expression enclosed in parentheses always results in only one value.
+local function enclosed_ret() return (ret2(1, 2)) end
+a,b = enclosed_ret()
+assert(a == 1 and b == nil)
+
+local function enclosed_vararg(...) return (...) end
+a,b,c=enclosed_vararg(1, 2, 3)
+assert(a == 1 and b == nil and c == nil)
```
Comment (yuin) [inline _lua5.1-tests/calls.lua:225]: This file is a copy of c-lua source code. Could you add these cases to `_glua-tests/issues.lua`?

**Review by yuin — CHANGES_REQUESTED** [review]

File: _lua5.1-tests/calls.lua (hunk 1 above)
Comment (0x501D) [reply _lua5.1-tests/calls.lua]: Done

**Comment by yuin** [comment]
It seems conflicts. Could you resolve it?

File: _lua5.1-tests/calls.lua (hunk 1 above)
Comment (0x501D) [reply _lua5.1-tests/calls.lua]: Added one more fix, rebased in single commit.

**Comment by 0x501D** [comment]
> It seems conflicts. Could you resolve it?

Done.
btw. now tests failed FAIL: TestTableLenType (0.00s)

**Pull request merged**